### DISCUSSION TITLE
Requesting New Photo app Immich to be added

### DISCUSSION THREAD

**Discussion by Dvalin21** [tteck/Proxmox#1668] [Ideas]
Requesting for Immich to be added to your list 
https://github.com/immich-app/immich
Thanks in advance.

**Comment by DocDrydenn** [comment]
From what I can tell, the main reason Immich (and some other requested apps) haven't been added already, is because they're only released via Docker/Snap/etc... They offer no other build options. My guess is if they offered some form of build-from-source or bare-metal install, then these apps could probably be added very quickly and easily to this repo. Conversion from docker to bare-metal doesn't seem to be a thing.

Anyway, just a guess on my part... seems to be the commonality of most of the pending requests vs. the requests that get added.

In the mean time, I've hit up the specific repos for the apps I'd like to see added here - requesting they offer up more build options. Example post for Immich [here](https://github.com/immich-app/immich/discussions/2968). (not my post, but I did add my vote and comments to it)

**Reply by MoldyTaint** [reply]
This was my feature request in their app page. I am in the same boat hoping there will be an easier way to run it on Proxmox. Tteck makes such great interactive installation scripts that i've been able to replace most of my docker services with the scripts alone.

**Comment by Dvalin21** [comment]
Thanks DocDrydenn, I appreciate the response.

**Comment by tteck** [comment]
https://github.com/immich-app/immich/discussions/1657

**Reply by oscarsantillana** [reply]
https://github.com/arter97/immich-native

**Comment by mastan30** [comment]
Here's a way to install immich in proxmox using docker unpriviliged LXC:


# Immich Docker Install

1. To install immich, the recommended way to install is using a docker compose, we can't install it on a bare metal debian os. So we will install a docker LXC using the proxmox script and install docker compose.Then we will create the docker-compose file for installing immich.

2. To install docker LXC we can used the following Proxmox script from the beloved tteck github:

		

		Docker LXC
		Options to Install Portainer and/or Docker Compose V2
		Docker is an open-source project for automating the deployment of applications as portable, self-sufficient containers.

		🛈 If the LXC is created Privileged, the script will automatically set up USB passthrough.

		To create a new Proxmox VE Docker LXC, run the command below in the Proxmox VE Shell.

		Copy
		bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/docker.sh)"
		⚡ Default Settings: 2GB RAM - 4GB Storage - 2vCPU ⚡
		As an alternative option, you can use Alpine Linux and the Docker package to create a Docker LXC container with faster creation time and minimal system resource usage.

		To create a new Proxmox VE Alpine-Docker LXC, run the command below in the Proxmox VE Shell.

		Copy
		bash -c "$(wget -qO - https://github.com/tteck/Proxmox/raw/main/ct/alpine-docker.sh)"
		⚡ Default Settings: 1GB RAM - 2GB Storage - 1vCPU ⚡
		⚠ Run Compose V2 by replacing the hyphen (-) with a space, using docker compose, instead of docker-compose.

		Portainer Interface: IP:9443


3. Once the container is installed, let's mount a mountpoint of SMB share where the images are uploaded.

4. In order to write the files into the SMB share, there's a catch here. The LXC we created using the script is an unpriviliged container which is the safest way to build container but the issue is we can't write anything into the shared drived from Proxmox host.

5. Instead of mounting the SMB share via Proxmox UI, we use fstab to mount the SMB share. Here are the steps to mount

	```
	  vi /etc/fstab
	```
	
	Then add the following line replacing the uid and gid for which you want the access to the proxmox container
	
	```
	  //192.168.X.XX/Path/to/share/ /path/to/mount/       cifs    noperm,iocharset=utf8,rw,credentials=/root/.storage_credentials,uid=1000,gid=1000,file_mode=0660,dir_mode=0770 0 0
	```
     eg: //192.168.1.19/Photos/immichtest/ /mnt/pve/immichtest/       cifs    noperm,iocharset=utf8,rw,credentials=/root/.storage_credentials,uid=1000,gid=1000,file_mode=0660,dir_mode=0770 0 0
	 
	```
      vi /root/.storage_credentials
    ```	
	 In this file add the username and password as below:
	 
	```
	username=username
	password=password
	```
	
	```
	mount -a
	```
   This will mount the SMB share into the Proxmox host, now check the mount directory permissions by doing ls -al and check if the mount has the permissions of assigned uid and gid. If not do a reboot of the host and it takes care of it.
   
6. Shutdown the container and mount the SMB share to the proxmox LXC by following commands:

   ```
    pct set 100 -mp0 /mnt/bindmounts/shared,mp=/shared
   
   ```
    eg: pct set 100 -mp0 /mnt/pve/immichtest/ ,mp=/immichtest/   - 100 is the lxc id
   
7. Start the container and check the if the mount point is assigned with right uid and gid inside the container.

8. Now in the LXC create a folder using following commands

   ```
     mkdir ./immich/
	 cd ./immich/
   ```
   
   create the docker-compose file and also the .env file in the directory as per the immich guide
   
   ```
      https://immich.app/docs/install/docker-compose
   ```
   
   Run the docker compose and follow the immichi guide to setup your albums and libraries as required.
   
   
9. Optinally in order to add hardware acceleration, I used NVIDIA acceleration. In order to get it working here are the steps to follow:
   
   Next step is to passthrough the nvidia graphics card to the lxc, In order to do the passthrough follow the below guide.
   
   > [Nvidia graphics passthrough](https://gist.github.com/egg82/90164a31db6b71d36fa4f4056bbee2eb_
   > [Nivdia graphics passthrough video](https://www.youtube.com/watch?v=-Us8KPOhOCY)
   
   I have already installed NVIDIA drivers for jellyfin lxc in proxmox host, so I just installed the nvidia drivers in container using below commands:
   
   ```
    wget https://us.download.nvidia.com/XFree86/Linux-x86_64/535.146.02/NVIDIA-Linux-x86_64-{Latest-version}.run
   ```
    you can get the NVIDIA latest driver url from here : https://www.nvidia.com/Download/index.aspx
	
   once downloaded install the drivers (The --no-kernel-module module is used as already there's kernel installed ):
   ```
   chmod +x NVIDIA-Linux-x86_64-{Latest-version}.run
   
   ./NVIDIA-Linux-x86_64-{Latest-version}.run ----no-kernel-module
   ```
   
   once installed make sure the drivers are installed by running:
   ```
   nvidia-smi -l 10
   ```
   
10. Now we need to pass through the hardware acceleration to docker container by following the immich guide:
    
	```
	https://immich.app/docs/features/hardware-transcoding
    ```
	
	In order to have the driver passed to container, we need to have nvidia container toolkit installed using the following commands in the container:
	(https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html)
	
	```
	curl -fsSL https://nvidia.github.io/libnvidia-container/gpgkey | sudo gpg --dearmor -o /usr/share/keyrings/nvidia-container-toolkit-keyring.gpg \
    && curl -s -L https://nvidia.github.io/libnvidia-container/stable/deb/nvidia-container-toolkit.list | \
    sed 's#deb https://#deb [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg] https://#g' | \
    sudo tee /etc/apt/sources.list.d/nvidia-container-toolkit.list
	
	
	
	sed -i -e '/experimental/ s/^#//g' /etc/apt/sources.list.d/nvidia-container-toolkit.list
	
	
	
	sudo apt-get update
	
	
	sudo apt-get install -y nvidia-container-toolkit
	
	
	```
	
	Also one other step I needed to do to make sure the docker compose works with hardware-transcoding is to modify nvidia config using below commands in container:
	
	Edit /etc/nvidia-container-runtime/config.toml and disable cgroups:
	```
      
	  vi /etc/nvidia-container-runtime/config.toml
	  
	  set no-cgroups = true in the config.toml file, save and reboot the container. Now run the docker compose for immich.

	```
	
	
	
	References:
	
	https://immich.app/docs/install/script
	
	https://forum.proxmox.com/threads/how-to-automatically-mount-a-smb-share-on-an-lxc.118220/

    https://immich.app/docs/features/hardware-transcoding
	
	https://www.reddit.com/r/Proxmox/comments/s0ud5y/comment/jl4lef2/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button
	
	<blockquote class="reddit-embed-bq" data-embed-height="348"><a href="https://www.reddit.com/r/Proxmox/comments/s0ud5y/comment/jl4lef2/">Comment</a><br> by<a href="https://www.reddit.com/user/Masterofchicken/">u/Masterofchicken</a> from discussion<a href="https://www.reddit.com/r/Proxmox/comments/s0ud5y/cgroups2_problem_with_nvidiacontainercli/"><no value=""></no></a><br> in<a href="https://www.reddit.com/r/Proxmox/">Proxmox</a></blockquote><script async="" src="https://embed.reddit.com/widgets.js" charset="UTF-8"></script>

**Reply by tteck** [reply]
![Screenshot 2023-12-10 6 08 10 PM](https://github.com/tteck/Proxmox/assets/21342670/e74cb838-c9cf-427d-b02c-ee56a4781213)
